### DISCUSSION TITLE
Shelly EM - strano comportamento sui valori riportati

### DISCUSSION THREAD

**Discussion by liuk4friends** [indomus/forum#1751] [GENERALE - Discussioni generiche]
Buonasera, ragazzi. 
Sono in una casa in affitto ove da tempo monitoro i consumi elettrici (per evitare i continui distacchi causati da quella sbadata di mia moglie) attraverso una serie di smart plug con EM (di varie marche, tutte Tuya comunque).
Non potendo avere un calcolo preciso principalmente per via del forno e della lavastoviglie non monitorati (ma anche diverse altre cose qua e là), ho deciso di passare a Shelly EM.

L'impianto è organizzato con quattro magnetotermici a valle del salvavita, ovvero:
- luci (10A);
- prese (16A);
- settore cucina;
- impianto aria condizionata (2 motori con 2 splitter ognuno).

Poco più di una settimana fa ho montato uno Shelly EM per monitorare il consumo del settore cucina (frigo, lavastoviglie, forno, cappa e un paio di prese sul piano) e dell'impianto di aria condizionata. 
Ho integrato lo Shelly EM in HA con MQTT e tutto funziona perfettamente.

Stasera ho montato una secondo Shelly EM per monitorare invece le luci e le prese. 
Stavolta però sono emersi due problemi. 

1) La pinza toroidale collegata alla fase delle luci mi restituisce un valore negativo di circa -4,5W, abbastanza costante, anche quando le luci sono tutte spente. 
Se accendo una lampadina da 8 W il consumo segna circa 4W. In pratica è come se il ch1 fosse starato e invece di partire da 0W quando tutto è spento, partisse appunto da circa -4,5W.
La cosa è ancora più strana tenendo conto che ho una lampadina smart Philips HUE zigbee (colore bianco) da 9,5W collegata alla linea luci. In stand-by dovrebbe consumare 0,5W, che è davvero poco, ma a maggior ragione non si giustifica un valore negativo da parte della pinza.

2)Il secondo problema mi sembra più grave. La seconda pinza toroidale (ch2) che monitora tutte le prese (tranne la cucina che, come detto, monitoro con il primo Shelly EM) restituisce valori che ballano troppo. 
L'oscillazione restituisce valori che cambiano ogni secondo da poche decine di Watt ad alcune centinaia, pur senza accendere o spegnere nulla (in casa ci sono alcuni dispositivi costantemente accesi, come un NAS, un Mac Mini che uso come server e qualche altra cosa qua e là).  
Devo sottolineare che alcuni dei dispositivi che ho elencato nella parentesi precedente sono separatamente monitorati anche da singole smartplug 16A (Tuya) e i valori, in quel caso, sono molto stabili. La somma di queste ultime inoltre è tendenzialmente più bassa di quello che mi fa vedere il ch2 dello Shelly EM. 
Insomma a me sembra che i valori riportati dallo Shelly EM siano sballati.

Preciso inoltre che:
1) ho utilizzato le pinze amperometriche della Shelly
2) ho settato correttamente la tipologia di pinza nella app Shelly (50A)
3) le ho montate nel verso giusto, seguendo il verso indicato K>L.

Siccome il quadro elettrico è molto piccolo e ho faticato non poco nel montaggio per via del poco spazio disponibile, ho provato anche a spostare le due pinze dalla fase in uscita a quella fase in ingresso dei due magnetotermici, rispettivamente luci e prese.

La situazione sostanzialmente non è cambiata, anzi forse è peggiorata. La pinza ch2 (che monitora le prese) continua ogni secondo a dare valori che oscillano di parecchie centinaia di W. La pinza ch1 (che monitora le luci) ora restituisce un valore negativo che oscilla da -2W a -4,6W (prima era più stabile sul -4,5W).

In pratica, chiedo a chi ha più esperienza di me cosa dovrei ricontrollare o cosa potrei fare per capire/risolvere il problema.

Aggiungo un'ultima cosa. In casa ho tre Fritz!Powerline per collegare la zona TV e la zona server (NAS e Mac mini) al router all'ingresso di casa. Chissà che non c'entrino qualcosa ....

Grazie in anticipo,
Luca

**Comment by liuk4friends** [comment]
Buonasera a tutti,
stasera ho fatto un test empirico per cercare di capire qualcosa in più.

Ho smontato lo Shelly EM e le due pinze dal quadro elettrico. 
Quindi ho collegato lo Shelly EM a una spina volante e ho fatto qualche prova con la fase di una lampadina basso consumo da 12 W.

La pinza sul canale ch1 sembra funzionare decentemente anche se non è molto precisa. 
Segna 10,02W invece dei 12W della lampadina.

La pinza sul canale ch2 è un vero disastro. Se non la collego ad una fase segna 33,5W. 
Quando la collego alla fase della lampadina mi segna oltre 2300W  !!!!

Dunque, posto che almeno un componente è rotto, ci sono delle prove che posso fare per capire quale sia guasto tra lo Shelly e la pinza?

grazie a tutti

**Comment by liuk4friends** [comment]
Ho avuto risposta sul canale Telegram.
Fatta ulteriore prova, invertendo le due pinze. Ancora valori sballati per di più con segno negativo.
Eseguito un factory reset, mi sono collegato al server web interno e i valori ancora fuori scala (stavolta in positivo).
Su consiglio di altri utenti del canale Telegram ho avviato il reso con Amazon.

**Comment by liuk4friends** [comment]
Dopo aver effettuato il reso su Amazon dello shelly guasto, è arrivato quello nuovo.
Dopo aver verificato il corretto funzionamento con la famosa lampadina da 12W, l'ho montato nel quadro elettrico.
Tutto ok.


![Screeny Shot 8 Jul 2022 at 00 07 24](https://user-images.githubusercontent.com/63516851/177879758-365ee21a-b049-4800-912b-a3329cf36108.png)

Penso si possa chiudere questo thread.

**Comment by directy01** [comment]
Molto bella questa plancia.. che tipo di card hai usato?
